### PR TITLE
Second step of the migration to not_null<>, covering geometry, integrators and physics

### PR DESCRIPTION
**Pull request by pleroy**

### CODE REVIEW

File: base/not_null.hpp [hunk 1]
```diff
@@ -181,6 +181,12 @@ class not_null {
   template<typename P = pointer, typename = decltype(std::declval<P>().get())>
   not_null<decltype(std::declval<P>().get())> get() const;
 
+  // When |pointer| has a |release()| member function, this returns
+  // |pointer_.release()|.
```
Comment (eggrobin) [inline base/not_null.hpp]: The `not_null<pointer>` is invalidated.

File: physics/n_body_system_test.cpp [hunk 2]
```diff
@@ -161,8 +163,8 @@ TEST_F(NBodySystemDeathTest, IntegrateError) {
                         trajectory1_.get()});
   }, "Multiple trajectories");
   EXPECT_DEATH({
-    std::unique_ptr<Trajectory<EarthMoonOrbitPlane>> trajectory =
-        std::make_unique<Trajectory<EarthMoonOrbitPlane>>(check_not_null(&body2_));
+    not_null<std::unique_ptr<Trajectory<EarthMoonOrbitPlane>>> trajectory =
```
Comment (eggrobin) [inline physics/n_body_system_test.cpp]: auto?

File: base/not_null.hpp (hunk 1 above)
Comment (pleroy) [inline base/not_null.hpp]: Done.

File: physics/n_body_system_test.cpp (hunk 2 above)
Comment (pleroy) [inline physics/n_body_system_test.cpp]: Done.

**Pull request merged**